### DISCUSSION TITLE
Recognize Clipping - complicated

### DISCUSSION THREAD

**Discussion by ciochu** [ImageMagick/ImageMagick#6212] [Consulting]
With my short journey with ImageMagick, and long one with Photoshop and Tiff files, I realized that clipping paths are not that easy topic to solve with any straightforward code. 

There are some files, where clipping path is hard to recognize (no name, or saved in awkward way) by ExifTool or ImageMagick 
Even Photophea (online ps) can not see reference to clipping path. 

I realized that by running clipping algorythm with ImageMagick and several other tools, and I realized that some TIFFs and JPEGs were successfully converted to transparent PNG even if I couldn't see clipping path reference in any of previously mentioned tools. 

I start testing various other solution like 
`for %f in (*.tiff) do (magick mogrify -path converted -format png -clip "%f" identify -format "%%[opaque]" converted\%~nf.png | findstr /i "false" >nul && ren converted\%~nf.png %~nf_clipped.png)`

but results are not good yet. 

I wanted to ask for your help in that matter. It would be complex, I'm happy to use paid consulting root, but I hope it's just me missing sth. (i checked multiple forums, asked GPT and experienced people)

Final goal is to be able to easily identify if files in a given folder have clipping path and write this information into EXIF field 
or list in log.txt file. 

It seems to be easy task but once you try to do so with that file, you realize that clipping path is clearly here (PNG transparent) , but system can not identify, name or existence of this path. 
https://images.salsify.com/image/upload/s--h7m0zwd2--/fl_attachment:1072828_-_VHB_5962/e7ba81e687ca1aca42e8000b7a5de7a74d0d08e5

**Comment by snibgo** [comment]
> It seems to be easy task but once you try to do so with that file, you realize that clipping path is clearly here (PNG transparent) , but system can not identify, name or existence of this path.

That file `1072828_-_VHB_5962.tiff` contains two images. "`magick identify -verbose`" clearly shows that both images have clipping paths.

**Reply by ciochu** [reply]
Thank you @snibgo , is there any chance to do simple testing for Clipping path? I know how to diplay this parapeter, but how to return "yes" if clipping path exist ?

**Reply by fire1** [reply]
> Thank you @snibgo , is there any chance to do simple testing for Clipping path? I know how to display this parameter, but how to return "yes" if clipping path exist ?

Mey, I try to help...
```
identify -quiet -format "%[8BIM:1999,2998:#1]" test.jpg
```
This command should output the first clipping path in an image "test.jpg",(to check for other change  `#1` to `#2` and so on..)  IM is converting it to SVG, from what I know you cannot rely on any other information such as clipping-path name or others. 


There is one huge problem with clipping paths, there is no tool or library to work reliably with them... I'm struggling for years with problems with those clipping paths, only thing left to do is blame Adobe... 

Moving, copying, removing or even existing clipping path ... is such a nightmare!

**Reply by ciochu** [reply]
@fire1  Thank you so much with sharing your experience. I'm not that advanced but I'll consult this with our developers. I will put here my findings for others.

**Comment by fire1** [comment]
Besides, working with clipping paths I'm recommending ExifTool, here is how you list clipping paths with a Photoshop config for that. 

Example: 
```
exiftool -config <photoshop_paths.config> ./masked.jpg -allpaths
```
It will print a list with Clipping paths names:

```
Path 7d0                        : Path 1
```
First is the clipping path index (`7d0`), "Path 1" is shown when is possible.
Here you can find  find [photoshop_paths.config](https://github.com/exiftool/exiftool/blob/master/config_files/photoshop_paths.config)

**Comment by MrMadClown** [comment]
I've been using both these methods for a while.

But, as it turns out, both do not distinguish between WorkingPaths and ClippingPaths.
For further processing of the JPEG I need the name of the clipping path, if I provide the name exifool returns it crashes
likewise identify returns the SVG as `Clipping Path:..` 

Does any one of you know of a Way to only get one of the two?
Or of a way to confirm if `Path X` is actually a clipping Path.
